src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeRanges.cs

```diff
@@ -32,9 +32,8 @@ private static UnicodeRange CreateEmptyRange(ref UnicodeRange range)
         {
             // If the range hasn't been created, create it now.
             // It's ok if two threads race and one overwrites the other's 'range' value.
-            var newRange = new UnicodeRange(0, 0);
-            Volatile.Write(ref range, newRange);
-            return newRange;
+            var range = new UnicodeRange(0, 0);
+            return range;
```

Comment:
> CLR must ensure partially instantiated objects are not visible outside of their ctors

Yes, on the same thread.

> the assignment is atomic as it's a reference type

That just means the reference itself won't get torn.

This might be fine, as the CLR does add extra fences to provide a stronger memory model than the CLI spec, but it'd be good to verify. There's a reason Volatile.Write exists ;)